specification/data-database.md

```diff
@@ -1,32 +1,75 @@
+executed on the database. It may be stored procedure name (without argument), SQL
 statement without variable arguments, etc. When it's impossible to get any
 meaningful representation of the span `name`, it can be populated using the same
-value as `db.instance`.
+value as `db.name`.
+
+## Connection-level attributes
 
-Note, Redis, Cassandra, HBase and other storage systems may reuse the same
-attribute names.
+These attributes will usually be the same for all operations performed over the same database connection
+although some database systems may allow to switch to a different `db.user` within the same connection
+and other database systems may not even have the concept of a connection.
 
 | Attribute name | Notes and examples                                           | Required? |
 | :------------- | :----------------------------------------------------------- | --------- |
-| `component`    | Database driver name or database name (when known) `"JDBI"`, `"jdbc"`, `"odbc"`, `"postgreSQL"`. | Yes       |
-| `db.type`      | Database type. For any SQL database, `"sql"`. For others, the lower-case database category, e.g. `"cassandra"`, `"hbase"`, or `"redis"`. | Yes       |
-| `db.instance`  | Database instance name. E.g., In java, if the jdbc.url=`"jdbc:mysql://db.example.com:3306/customers"`, the instance name is `"customers"`. | Yes       |
-| `db.statement` | A database statement for the given database type. Note, that the value may be sanitized to exclude sensitive information. E.g., for `db.type="sql"`, `"SELECT * FROM wuser_table"`; for `db.type="redis"`, `"SET mykey 'WuValue'"`. | Yes       |
+| `component` | Always the string `"db"` | Yes       |
+| `db.type` | The lower-case database category, e.g. `"sql"`, `"cassandra"`, `"hbase"`, or `"redis"`. | Yes       |
+| `db.tech` | Name of the database technology used, e.g. "SQL Server" or "MySQL". | No     |
+| `db.url` | The connection string used to connect to the database | No       |
 | `db.user`      | Username for accessing database. E.g., `"readonly_user"` or `"reporting_user"` | No        |
+| `db.portname` | A name that identifies the `net.peer.port` used to connect. This property is intended to be used for [MS SQL Server instances][] but can be used for any database system that provides a name to port mapping. | See below        |
+
+[MS SQL Server instances]: https://docs.microsoft.com/en-us/sql/connect/jdbc/building-the-connection-url?view=sql-server-ver15
+
+Additionally at least one of `net.peer.name` or `net.peer.ip` from the [network attributes][] is required and `net.peer.port` is recommended.
+If using a non-standard port for the `db.tech`, at least one of `net.peer.port` or `db.portname` is required.
+
+[network attributes]: data-span-general.md#general-network-attributes
+
+## Call-level attributes
+
+These attributes may be different for each operation performed, even if it uses the same connection
```

Comment:
>These attributes may be different for each operation performed, even if it uses the same connection although usually only one db.name will be used per connection.

These attributes may be different for each operation performed even if the same connection is used for multiple operations. Usually, only one `db.name` will be used per connection.